src/datanode/src/datanode.rs

```diff
@@ -334,6 +334,9 @@ impl DatanodeBuilder {
             common_runtime::global_runtime(),
             event_listener,
             table_provider_factory,
+            opts.max_concurrent_queries,
+            //TODO: revaluate the hardcoded timeout on the next version of datanode concurrency limiter.
+            Duration::from_millis(25),
```

Comment:
25ms seems to be too short. Maybe we need to hold those throttled queries instead of fail them very quickly.

Reply:
I can increase the threshold. By hold do you mean to allow queries to wait for the semaphore until the request to datanode timeout?